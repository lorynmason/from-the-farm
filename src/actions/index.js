export const addUser = (user) => {
  return {
    type: 'ADD_USER',
    user
  };
};

export const removeUser = () => {
  return {
    type: 'REMOVE_USER'
  };
};

export const addVendors = (vendors) => {
  return {
    type: 'ADD_VENDORS',
    vendors
  };
};

export const addProducts = (products) => {
  return {
    type: 'ADD_PRODUCTS',
    products
  };
};

export const isLoading = (bool) => {
  return {
    type: 'IS_LOADING',
    bool
  };
};

export const hasErrored = (message) => {
  return {
    type: 'HAS_ERRORED',
    message
  };
};

export const addMessage = (message) => {
  return {
    type: 'ADD_MESSAGE',
    message
  };
};

export const addItemList = (items) => {
  return {
    type: 'ADD_ITEM_LIST',
    items
  };
};

export const vendorSearchResults = (results) => {
  return {
    type: 'VENDOR_SEARCH',
    results
  };
};

export const productSearchResults = (results) => {
  return {
    type: 'PRODUCT_SEARCH',
    results
  };
};

export const clearSearchResults = () => {
  return {
    type: 'CLEAR_SEARCH'
  };
};
