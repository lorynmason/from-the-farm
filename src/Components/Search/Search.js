import React, { Component } from 'react';
import { connect } from 'react-redux';
import { addProducts } from '../../actions';

export class Search extends Component {
  constructor(){
    super()
    this.state = {
      productId: '',
      location: '',
      range: ''
    }
  }

  handleChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  sendSearch = (e) => {
    e.preventDefault()
    this.props.search(this.state)
  }

  render() {
    const productOptionsNames = this.props.products.reduce((arr, product) => {
      if(!arr.includes(product.name)) {
        arr.push(product.name)
      }
      return arr     
    },[]);

    const productOptionsIds = this.props.products.reduce((arr, product) => {
      if(!arr.includes(product.item_id)) {
        arr.push(product.item_id)
      }
      return arr     
    },[]);

    const productOptions = productOptionsNames.map((product, i) => {
        return <option value={productOptionsIds[i]} key={productOptionsIds[i]}>{product}</option>
    });

    return (
      <section className="search">
        <h3>Narrow Your Search</h3>
        <form className="search-form" onSubmit={this.sendSearch} onChange={this.handleChange}>
          <div className="search-options">
            <select id="product-options" name="productId">
            <option value="">select a product</option>
              { productOptions }
            </select>
            <input placeholder="location, address, zipcode" name="location"/>
            <select id="radius-options" name="range">
              <option value="50">50 mile radius</option>
              <option value="100">100 mile radius</option>
              <option value="150">150 mile radius</option>
              <option value="200">200 mile radius</option>
              <option value="250">250 mile radius</option>
              <option value="300">300 mile radius</option>
              <option value="500">500 mile radius</option>
            </select>
          </div>
          <button>Search</button>
        </form>
      </section>
    )
  }
}

export const mapStateToProps = (state) => ({
  products: state.products
});

export const mapDispatchToProps = (dispatch) => {
  return {
    addProductsToStore: (products) => dispatch(addProducts(products))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Search);