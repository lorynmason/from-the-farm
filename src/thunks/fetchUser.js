import { isLoading, hasErrored, addUser } from '../actions';
import { cleanUser } from '../helpers/cleaner'

export const fetchUser = (url, token) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true));
      const response = await fetch(url);
      if (!response.ok) {
        throw Error(response.statusText);
      }
      dispatch(isLoading(false))
      const results = await response.json();
      const user = cleanUser(results.data, token);
      dispatch(addUser(user));
    } catch (error) {
      console.log(error)
      dispatch(hasErrored(error.message));
    }
  }
}