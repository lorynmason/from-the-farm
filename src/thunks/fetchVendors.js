import { isLoading, hasErrored, addVendors, addProducts } from '../actions';

export const fetchVendors = () => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true));
      const response = await fetch('https://xpoll-be.herokuapp.com/api/v1/vendors');
      if (!response.ok) {
        throw Error(response.statusText);
      }
      dispatch(isLoading(false))
      const results = await response.json();
      const vendors = cleanVendors(results.data);
      const products = cleanProducts(results.data);
      dispatch(addVendors(vendors));
      dispatch(addProducts(products));
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}

export const cleanVendors = (vendors) => {
  return vendors.map( vendor => {
    const { name, address, state, city, phone, email, lat, long, bio } = vendor.attributes
    return {
      name,
      bio,
      address,
      state, 
      city,
      phone,
      email,
      lat,
      long, 
      id: parseInt(vendor.id)
    }
  })
}
  
export const cleanProducts = (vendors) => {
  return vendors.reduce((productArr, vendor) => {
    const { name, products } = vendor.attributes;
    products.reduce((obj, product) => {
      const itemName = Object.keys(product)[0];

      obj = {
        vendorName: name,
        name: Object.keys(product)[0],
        ...product[itemName]
      }
      productArr.push(obj);
      return obj
    }, {})

    return productArr;
  }, [])
}
