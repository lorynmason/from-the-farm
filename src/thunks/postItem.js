import { isLoading, addMessage, hasErrored } from '../actions';

export const postItem = (url, item, userToken) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true));
      const response = await fetch(url, {
        method: 'POST',
        headers: {
          "Content-Type": "application/json",
          "Accept": "application/json",
          "Authorization": `Bearer ${userToken}`
        },
        body: JSON.stringify(item)
      });
      if (!response.ok) {
        throw Error(response.statusText);
      }
      dispatch(isLoading(false));
      const results = await response.json();
      const newItem = results.data.attributes;
      dispatch(addMessage(`Successfully added ${newItem.name} at ${newItem.price} per ${newItem.unit}`));
    } catch(error) {
      dispatch(hasErrored(error.message));
      dispatch(addMessage('Unable to add product'));
    }
  }
}