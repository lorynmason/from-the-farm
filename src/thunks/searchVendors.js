import { isLoading, hasErrored, vendorSearchResults, productSearchResults, addMessage } from '../actions';
import { cleanVendors, cleanProducts } from '../helpers/cleaner'

export const searchVendors = (url, productId) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true));
      const response = await fetch(url);
      if (!response.ok) {
        throw Error(response.statusText);
      }
      dispatch(isLoading(false))
      const results = await response.json();
      console.log(results)
      if(!results.data.length) {
        dispatch(addMessage('No Search Results'))
      }
      const ids = results.data.reduce((arr, result) => {
        if(!arr.includes(result.id)) {
          arr.push(result.id)
        }
        return arr
      },[])
      const matches = ids.map(id => results.data.find(result => result.id === id))
      let vendors = cleanVendors(matches);
      let products = cleanProducts(matches);
      vendors = vendors.map(vendor => vendor.id)
      if(productId) {
        products = products.filter(product => {
          return product.item_id === parseInt(productId)
        }).map(product => product.id)
      } else {
        products = products.map(product => product.id)
      }
      dispatch(vendorSearchResults(vendors))
      dispatch(productSearchResults(products))
    } catch (error) {
      console.log(error)
      dispatch(hasErrored(error.message));
    }
  }
}