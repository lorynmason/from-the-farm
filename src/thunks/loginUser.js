import { isLoading, hasErrored } from '../actions';
import { fetchUser } from './fetchUser'

export const loginUser = (url, state) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true));
      const response = await fetch(url, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json'
        },
        body: JSON.stringify({
          email: state.email,
          password: state.password
        })
      });
      if (!response.ok) {
        throw Error(response.statusText);
      }
      dispatch(isLoading(false))
      const results = await response.json();
      dispatch(fetchUser(`https://xpoll-be.herokuapp.com/api/v1/users/${results.user.id}`, results.auth_token)) 
    } catch (error) {
      dispatch(hasErrored(error.message));
    }
  }
}