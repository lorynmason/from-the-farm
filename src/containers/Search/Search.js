import React, { Component } from 'react';
import { connect } from 'react-redux';
import { addProducts, addMessage, clearSearchResults } from '../../actions';
import { searchVendors } from '../../thunks/searchVendors';
import PropTypes from 'prop-types';

export class Search extends Component {
  constructor(){
    super()
    this.state = {
      productId: '',
      location: '',
      range: ''
    }
  }

  handleChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  handleSubmit = (e) => {
    const { productId, location, range } = this.state
    let newRange = range
    e.preventDefault()
    const baseUrl = 'https://xpoll-be.herokuapp.com/api/v1/search'
    let path;
    if (!location) {
      this.props.addMessage('error: please enter a search location')
      return
    } 
    if (!range) {
      newRange = "50"
    } 
    if (!productId) {
      path = `?loc=${location}&range=${newRange}`
    } else {
      path = `?loc=${location}&range=${newRange}&item=${productId}`
    }
    this.props.searchVendors(baseUrl + path, productId)
    this.setState({
      location: ''
    })
  }

  reset = () => {
    this.props.clearSearchResults()
  }

  render() {
    const productOptions = this.props.items.map((item) => {
      return <option value={item.id} key={item.id}>{item.name}</option>
    });

    return (
      <section className="search">
        <form className="search-form" onSubmit={this.handleSubmit} onChange={this.handleChange}>
        <h3>Narrow Your Search</h3>
          <div className="search-options">
            <select id="product-options" name="productId" value={this.state.productId} onChange={this.handleChange}>
            <option value="">select a product</option>
              { productOptions }
            </select>
            <input placeholder="location, address, zipcode" name="location" value={this.state.location} onChange={this.handleChange}/>
            <select id="radius-options" name="range" value={this.state.range} onChange={this.handleChange}>
              <option value="50">50 mile radius</option>
              <option value="100">100 mile radius</option>
              <option value="150">150 mile radius</option>
              <option value="200">200 mile radius</option>
              <option value="250">250 mile radius</option>
              <option value="300">300 mile radius</option>
              <option value="500">500 mile radius</option>
            </select>
          </div>
          <button>Search</button>
          <p type="button" id="reset" onClick={this.reset}>Reset</p>
        </form>
      </section>
    )
  }
}

export const mapStateToProps = (state) => ({
  products: state.products,
  items: state.items
});

export const mapDispatchToProps = (dispatch) => {
  return {
    addProductsToStore: (products) => dispatch(addProducts(products)),
    searchVendors: (url, id) => dispatch(searchVendors(url, id)),
    addMessage: (message) => dispatch(addMessage(message)),
    clearSearchResults: () => dispatch(clearSearchResults())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Search);

Search.propTypes = {
  products: PropTypes.array,
  items: PropTypes.array,
  addProductsToStore: PropTypes.func,
  searchVendors: PropTypes.func,
  addMessage: PropTypes.func,
  clearSearchResults: PropTypes.func
}
