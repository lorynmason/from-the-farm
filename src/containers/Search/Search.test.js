import React from 'react';
import { Search, mapStateToProps, mapDispatchToProps } from './Search';
import { shallow } from 'enzyme';
import { addProducts } from '../../actions';
import { searchVendors } from '../../thunks/searchVendors';

jest.mock('../../thunks/searchVendors.js');

describe('Search', () => {
  let items = [{name: "Berries",
  id: 1}, {name: "Eggs",
  id: 11}]
  let wrapper = shallow(<Search items={items} products={[]} addProductsToStore={jest.fn()} searchVendors={jest.fn()} addMessage={jest.fn()} />);

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('handleChange', () => {
    it('should update state when changes are made to form elements', () => {
      expect(wrapper.state()).toEqual({
        productId: '',
        location: '',
        range: ''
      });

      wrapper.find('.search-form').simulate('change', {'target': {name: 'productId', value: '1'}});
      wrapper.find('.search-form').simulate('change', {'target': {name: 'location', value: '80219'}});
      wrapper.find('.search-form').simulate('change', {'target': {name: 'range', value: '50'}});

      expect(wrapper.state()).toEqual({
        productId: '1',
        location: '80219',
        range: '50'
      });
    });
  });

  describe('handleSubmit', () => {
    it('should call search with the correct params on submit', () => {
      wrapper.setProps({searchVendors: jest.fn()});

      const { searchVendors } = wrapper.instance().props
      wrapper.find('.search-form').simulate('submit', {preventDefault: ()=> jest.fn()});

      expect(searchVendors).toHaveBeenCalled();
    });
  });

  describe('mapStateToProps', () => {
    it('should return an array of products', () => {
      const mockGlobalState = {
        vendors: [{id: 1, name: 'Farm Team'}],
        products: [{name: 'carrots', quantity: 'all'}],
        user: {name: 'Mike'}
      }

      const expected = [{name: 'carrots', quantity: 'all'}];

      const mappedProps = mapStateToProps(mockGlobalState);
      expect(mappedProps.products).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    it('should dispatch addProducts with the correct params when addProductsToStore is called', () => {
      const mockProducts = [{name: 'carrots', quantity: 'all'}];
      const expected = addProducts(mockProducts);

      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.addProductsToStore(mockProducts);

      expect(mockDispatch).toHaveBeenCalledWith(expected);
    });
  });
});