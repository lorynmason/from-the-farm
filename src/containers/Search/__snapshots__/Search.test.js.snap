// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    addMessage={[MockFunction]}
    addProductsToStore={[MockFunction]}
    items={
      Array [
        Object {
          "id": 1,
          "name": "Berries",
        },
        Object {
          "id": 11,
          "name": "Eggs",
        },
      ]
    }
    products={Array []}
    searchVendors={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="search-form"
        onChange={[Function]}
        onSubmit={[Function]}
      >
        <h3>
          Narrow Your Search
        </h3>
        <div
          className="search-options"
        >
          <select
            id="product-options"
            name="productId"
          >
            <option
              value=""
            >
              select a product
            </option>
            <option
              value={1}
            >
              Berries
            </option>
            <option
              value={11}
            >
              Eggs
            </option>
          </select>
          <input
            name="location"
            placeholder="location, address, zipcode"
            value=""
          />
          <select
            id="radius-options"
            name="range"
          >
            <option
              value="50"
            >
              50 mile radius
            </option>
            <option
              value="100"
            >
              100 mile radius
            </option>
            <option
              value="150"
            >
              150 mile radius
            </option>
            <option
              value="200"
            >
              200 mile radius
            </option>
            <option
              value="250"
            >
              250 mile radius
            </option>
            <option
              value="300"
            >
              300 mile radius
            </option>
            <option
              value="500"
            >
              500 mile radius
            </option>
          </select>
        </div>
        <button>
          Search
        </button>
        <p
          id="reset"
          onClick={[Function]}
          type="button"
        >
          Reset
        </p>
      </form>,
      "className": "search",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Narrow Your Search
          </h3>,
          <div
            className="search-options"
          >
            <select
              id="product-options"
              name="productId"
            >
              <option
                value=""
              >
                select a product
              </option>
              <option
                value={1}
              >
                Berries
              </option>
              <option
                value={11}
              >
                Eggs
              </option>
            </select>
            <input
              name="location"
              placeholder="location, address, zipcode"
              value=""
            />
            <select
              id="radius-options"
              name="range"
            >
              <option
                value="50"
              >
                50 mile radius
              </option>
              <option
                value="100"
              >
                100 mile radius
              </option>
              <option
                value="150"
              >
                150 mile radius
              </option>
              <option
                value="200"
              >
                200 mile radius
              </option>
              <option
                value="250"
              >
                250 mile radius
              </option>
              <option
                value="300"
              >
                300 mile radius
              </option>
              <option
                value="500"
              >
                500 mile radius
              </option>
            </select>
          </div>,
          <button>
            Search
          </button>,
          <p
            id="reset"
            onClick={[Function]}
            type="button"
          >
            Reset
          </p>,
        ],
        "className": "search-form",
        "onChange": [Function],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Narrow Your Search",
          },
          "ref": null,
          "rendered": "Narrow Your Search",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <select
                id="product-options"
                name="productId"
              >
                <option
                  value=""
                >
                  select a product
                </option>
                <option
                  value={1}
                >
                  Berries
                </option>
                <option
                  value={11}
                >
                  Eggs
                </option>
              </select>,
              <input
                name="location"
                placeholder="location, address, zipcode"
                value=""
              />,
              <select
                id="radius-options"
                name="range"
              >
                <option
                  value="50"
                >
                  50 mile radius
                </option>
                <option
                  value="100"
                >
                  100 mile radius
                </option>
                <option
                  value="150"
                >
                  150 mile radius
                </option>
                <option
                  value="200"
                >
                  200 mile radius
                </option>
                <option
                  value="250"
                >
                  250 mile radius
                </option>
                <option
                  value="300"
                >
                  300 mile radius
                </option>
                <option
                  value="500"
                >
                  500 mile radius
                </option>
              </select>,
            ],
            "className": "search-options",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    value=""
                  >
                    select a product
                  </option>,
                  Array [
                    <option
                      value={1}
                    >
                      Berries
                    </option>,
                    <option
                      value={11}
                    >
                      Eggs
                    </option>,
                  ],
                ],
                "id": "product-options",
                "name": "productId",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "select a product",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": "select a product",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Berries",
                    "value": 1,
                  },
                  "ref": null,
                  "rendered": "Berries",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "11",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Eggs",
                    "value": 11,
                  },
                  "ref": null,
                  "rendered": "Eggs",
                  "type": "option",
                },
              ],
              "type": "select",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "location",
                "placeholder": "location, address, zipcode",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    value="50"
                  >
                    50 mile radius
                  </option>,
                  <option
                    value="100"
                  >
                    100 mile radius
                  </option>,
                  <option
                    value="150"
                  >
                    150 mile radius
                  </option>,
                  <option
                    value="200"
                  >
                    200 mile radius
                  </option>,
                  <option
                    value="250"
                  >
                    250 mile radius
                  </option>,
                  <option
                    value="300"
                  >
                    300 mile radius
                  </option>,
                  <option
                    value="500"
                  >
                    500 mile radius
                  </option>,
                ],
                "id": "radius-options",
                "name": "range",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "50 mile radius",
                    "value": "50",
                  },
                  "ref": null,
                  "rendered": "50 mile radius",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "100 mile radius",
                    "value": "100",
                  },
                  "ref": null,
                  "rendered": "100 mile radius",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "150 mile radius",
                    "value": "150",
                  },
                  "ref": null,
                  "rendered": "150 mile radius",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "200 mile radius",
                    "value": "200",
                  },
                  "ref": null,
                  "rendered": "200 mile radius",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "250 mile radius",
                    "value": "250",
                  },
                  "ref": null,
                  "rendered": "250 mile radius",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "300 mile radius",
                    "value": "300",
                  },
                  "ref": null,
                  "rendered": "300 mile radius",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "500 mile radius",
                    "value": "500",
                  },
                  "ref": null,
                  "rendered": "500 mile radius",
                  "type": "option",
                },
              ],
              "type": "select",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search",
          },
          "ref": null,
          "rendered": "Search",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Reset",
            "id": "reset",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Reset",
          "type": "p",
        },
      ],
      "type": "form",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="search-form"
          onChange={[Function]}
          onSubmit={[Function]}
        >
          <h3>
            Narrow Your Search
          </h3>
          <div
            className="search-options"
          >
            <select
              id="product-options"
              name="productId"
            >
              <option
                value=""
              >
                select a product
              </option>
              <option
                value={1}
              >
                Berries
              </option>
              <option
                value={11}
              >
                Eggs
              </option>
            </select>
            <input
              name="location"
              placeholder="location, address, zipcode"
              value=""
            />
            <select
              id="radius-options"
              name="range"
            >
              <option
                value="50"
              >
                50 mile radius
              </option>
              <option
                value="100"
              >
                100 mile radius
              </option>
              <option
                value="150"
              >
                150 mile radius
              </option>
              <option
                value="200"
              >
                200 mile radius
              </option>
              <option
                value="250"
              >
                250 mile radius
              </option>
              <option
                value="300"
              >
                300 mile radius
              </option>
              <option
                value="500"
              >
                500 mile radius
              </option>
            </select>
          </div>
          <button>
            Search
          </button>
          <p
            id="reset"
            onClick={[Function]}
            type="button"
          >
            Reset
          </p>
        </form>,
        "className": "search",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Narrow Your Search
            </h3>,
            <div
              className="search-options"
            >
              <select
                id="product-options"
                name="productId"
              >
                <option
                  value=""
                >
                  select a product
                </option>
                <option
                  value={1}
                >
                  Berries
                </option>
                <option
                  value={11}
                >
                  Eggs
                </option>
              </select>
              <input
                name="location"
                placeholder="location, address, zipcode"
                value=""
              />
              <select
                id="radius-options"
                name="range"
              >
                <option
                  value="50"
                >
                  50 mile radius
                </option>
                <option
                  value="100"
                >
                  100 mile radius
                </option>
                <option
                  value="150"
                >
                  150 mile radius
                </option>
                <option
                  value="200"
                >
                  200 mile radius
                </option>
                <option
                  value="250"
                >
                  250 mile radius
                </option>
                <option
                  value="300"
                >
                  300 mile radius
                </option>
                <option
                  value="500"
                >
                  500 mile radius
                </option>
              </select>
            </div>,
            <button>
              Search
            </button>,
            <p
              id="reset"
              onClick={[Function]}
              type="button"
            >
              Reset
            </p>,
          ],
          "className": "search-form",
          "onChange": [Function],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Narrow Your Search",
            },
            "ref": null,
            "rendered": "Narrow Your Search",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <select
                  id="product-options"
                  name="productId"
                >
                  <option
                    value=""
                  >
                    select a product
                  </option>
                  <option
                    value={1}
                  >
                    Berries
                  </option>
                  <option
                    value={11}
                  >
                    Eggs
                  </option>
                </select>,
                <input
                  name="location"
                  placeholder="location, address, zipcode"
                  value=""
                />,
                <select
                  id="radius-options"
                  name="range"
                >
                  <option
                    value="50"
                  >
                    50 mile radius
                  </option>
                  <option
                    value="100"
                  >
                    100 mile radius
                  </option>
                  <option
                    value="150"
                  >
                    150 mile radius
                  </option>
                  <option
                    value="200"
                  >
                    200 mile radius
                  </option>
                  <option
                    value="250"
                  >
                    250 mile radius
                  </option>
                  <option
                    value="300"
                  >
                    300 mile radius
                  </option>
                  <option
                    value="500"
                  >
                    500 mile radius
                  </option>
                </select>,
              ],
              "className": "search-options",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <option
                      value=""
                    >
                      select a product
                    </option>,
                    Array [
                      <option
                        value={1}
                      >
                        Berries
                      </option>,
                      <option
                        value={11}
                      >
                        Eggs
                      </option>,
                    ],
                  ],
                  "id": "product-options",
                  "name": "productId",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "select a product",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": "select a product",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": "1",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Berries",
                      "value": 1,
                    },
                    "ref": null,
                    "rendered": "Berries",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": "11",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Eggs",
                      "value": 11,
                    },
                    "ref": null,
                    "rendered": "Eggs",
                    "type": "option",
                  },
                ],
                "type": "select",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "location",
                  "placeholder": "location, address, zipcode",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <option
                      value="50"
                    >
                      50 mile radius
                    </option>,
                    <option
                      value="100"
                    >
                      100 mile radius
                    </option>,
                    <option
                      value="150"
                    >
                      150 mile radius
                    </option>,
                    <option
                      value="200"
                    >
                      200 mile radius
                    </option>,
                    <option
                      value="250"
                    >
                      250 mile radius
                    </option>,
                    <option
                      value="300"
                    >
                      300 mile radius
                    </option>,
                    <option
                      value="500"
                    >
                      500 mile radius
                    </option>,
                  ],
                  "id": "radius-options",
                  "name": "range",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "50 mile radius",
                      "value": "50",
                    },
                    "ref": null,
                    "rendered": "50 mile radius",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "100 mile radius",
                      "value": "100",
                    },
                    "ref": null,
                    "rendered": "100 mile radius",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "150 mile radius",
                      "value": "150",
                    },
                    "ref": null,
                    "rendered": "150 mile radius",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "200 mile radius",
                      "value": "200",
                    },
                    "ref": null,
                    "rendered": "200 mile radius",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "250 mile radius",
                      "value": "250",
                    },
                    "ref": null,
                    "rendered": "250 mile radius",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "300 mile radius",
                      "value": "300",
                    },
                    "ref": null,
                    "rendered": "300 mile radius",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "500 mile radius",
                      "value": "500",
                    },
                    "ref": null,
                    "rendered": "500 mile radius",
                    "type": "option",
                  },
                ],
                "type": "select",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search",
            },
            "ref": null,
            "rendered": "Search",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Reset",
              "id": "reset",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "Reset",
            "type": "p",
          },
        ],
        "type": "form",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
